{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [],
   "source": [
    "diann = \"/home/ekvall/diann/diann-linux\"\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Using Fragpipe's labling each replicate in each group -> One spectral library per replicate"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Run Diann with each spectral library, with its corresponding raw file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [],
   "source": [
    "def runDiann(short_name, exp_name):\n",
    "    for r_n in range(1,6):\n",
    "        print(\"-------- Get quant for replicate {} in experiment group {}\".format(r_n, exp_name))\n",
    "        print(\"\\n\")\n",
    "        path = '/home/ekvall/Desktop/sample_A_and_S/{}_{}'.format(short_name, r_n)\n",
    "        lib = path + \"/library.tsv\"\n",
    "        out = path + \"/diann_out.tsv\"\n",
    "    \n",
    "        file = \"/sdd/proteomics/DATA/PXD014076/{}/mzML/{}_Proteome_DIA_RAW_{}0{}.mzML\".format(exp_name, exp_name, short_name, r_n)\n",
    "    \n",
    "        run = \" \".join([diann, \"--f\", file, \"--lib\", lib, \"--out\", out])\n",
    "    \n",
    "        !{run}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Run for each replicate in the Aposymb group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------- Get quant for replicate 1 in experiment group Aposymb\n",
      "\n",
      "\n",
      "DIA-NN 1.7.12 (Data Independent Acquisition by Neural Networks)\n",
      "Compiled on Feb 13 2021 17:30:41\n",
      "Current date and time: Mon Mar  1 12:42:17 2021\n",
      "Logical CPU cores: 8\n",
      "\n",
      "1 files will be processed\n",
      "[0:00] Loading spectral library /home/ekvall/Desktop/sample_A_and_S/A_1/library.tsv\n",
      "[0:00] Finding proteotypic peptides (assuming that the list of UniProt ids provided for each peptide is complete)\n",
      "[0:00] Spectral library loaded: 168 protein isoforms, 168 protein groups and 183 precursors in 183 elution groups.\n",
      "[0:00] Initialising library\n",
      "[0:00] Saving the library to /home/ekvall/Desktop/sample_A_and_S/A_1/library.tsv.speclib\n",
      "\n",
      "[0:00] File #1/1\n",
      "[0:00] Loading run /sdd/proteomics/DATA/PXD014076/Aposymb/mzML/Aposymb_Proteome_DIA_RAW_A01.mzML\n",
      "[0:24] 183 library precursors are potentially detectable\n",
      "[0:24] Processing...\n",
      "[0:27] RT window set to 2.37209\n",
      "[0:27] Peak width: 4.43646\n",
      "[0:27] Scan window radius set to 9\n",
      "[0:28] Optimised mass accuracy: 6.64533 ppm\n",
      "[0:28] Removing interfering precursors\n",
      "[0:28] Training the neural network: 183 targets, 135 decoys\n",
      "[0:28] Number of IDs at 0.01 FDR: 178\n",
      "[0:28] Too low number of IDs with NNs: reverting to the linear classifier\n",
      "[0:28] Number of IDs at 0.01 FDR: 177\n",
      "[0:28] Calculating protein q-values\n",
      "[0:28] Number of protein isoforms identified at 1% FDR: 162 (precursor-level), 162 (protein-level) (inference performed using proteotypic peptides only)\n",
      "[0:28] Quantification\n",
      "[0:28] Quantification information saved to /sdd/proteomics/DATA/PXD014076/Aposymb/mzML/Aposymb_Proteome_DIA_RAW_A01.mzML.quant.\n",
      "\n",
      "[0:28] Cross-run analysis\n",
      "[0:28] Reading quantification information: 1 files\n",
      "[0:28] Quantifying peptides\n",
      "[0:28] Assembling protein groups\n",
      "[0:28] Quantifying proteins\n",
      "[0:28] Calculating q-values for protein and gene groups\n",
      "[0:28] Writing report\n",
      "[0:28] Report saved to /home/ekvall/Desktop/sample_A_and_S/A_1/diann_out.tsv.\n",
      "[0:28] Stats report saved to /home/ekvall/Desktop/sample_A_and_S/A_1/diann_out.stats.tsv\n",
      "[0:28] Writing gene report\n",
      "[0:28] Gene report saved to report.genes.tsv.\n",
      "[0:28] Log saved to /home/ekvall/Desktop/sample_A_and_S/A_1/diann_out.log.txt\n",
      "Finished\n",
      "\n",
      "-------- Get quant for replicate 2 in experiment group Aposymb\n",
      "\n",
      "\n",
      "DIA-NN 1.7.12 (Data Independent Acquisition by Neural Networks)\n",
      "Compiled on Feb 13 2021 17:30:41\n",
      "Current date and time: Mon Mar  1 12:42:46 2021\n",
      "Logical CPU cores: 8\n",
      "\n",
      "1 files will be processed\n",
      "[0:00] Loading spectral library /home/ekvall/Desktop/sample_A_and_S/A_2/library.tsv\n",
      "[0:00] Finding proteotypic peptides (assuming that the list of UniProt ids provided for each peptide is complete)\n",
      "[0:00] Spectral library loaded: 155 protein isoforms, 155 protein groups and 166 precursors in 166 elution groups.\n",
      "[0:00] Initialising library\n",
      "[0:00] Saving the library to /home/ekvall/Desktop/sample_A_and_S/A_2/library.tsv.speclib\n",
      "\n",
      "[0:00] File #1/1\n",
      "[0:00] Loading run /sdd/proteomics/DATA/PXD014076/Aposymb/mzML/Aposymb_Proteome_DIA_RAW_A02.mzML\n",
      "[0:24] 166 library precursors are potentially detectable\n",
      "[0:24] Processing...\n",
      "[0:26] RT window set to 2.37389\n",
      "[0:26] Peak width: 4.58537\n",
      "[0:26] Scan window radius set to 10\n",
      "[0:27] Optimised mass accuracy: 6.30543 ppm\n",
      "[0:27] Removing interfering precursors\n",
      "[0:27] Training the neural network: 166 targets, 112 decoys\n",
      "[0:27] Number of IDs at 0.01 FDR: 166\n",
      "[0:27] Calculating protein q-values\n",
      "[0:27] Number of protein isoforms identified at 1% FDR: 155 (precursor-level), 0 (protein-level) (inference performed using proteotypic peptides only)\n",
      "[0:27] Quantification\n",
      "[0:27] Quantification information saved to /sdd/proteomics/DATA/PXD014076/Aposymb/mzML/Aposymb_Proteome_DIA_RAW_A02.mzML.quant.\n",
      "\n",
      "[0:27] Cross-run analysis\n",
      "[0:27] Reading quantification information: 1 files\n",
      "[0:27] Quantifying peptides\n",
      "[0:27] Assembling protein groups\n",
      "[0:27] Quantifying proteins\n",
      "[0:27] Calculating q-values for protein and gene groups\n",
      "[0:27] Writing report\n",
      "[0:28] Report saved to /home/ekvall/Desktop/sample_A_and_S/A_2/diann_out.tsv.\n",
      "[0:28] Stats report saved to /home/ekvall/Desktop/sample_A_and_S/A_2/diann_out.stats.tsv\n",
      "[0:28] Writing gene report\n",
      "[0:28] Gene report saved to report.genes.tsv.\n",
      "[0:28] Log saved to /home/ekvall/Desktop/sample_A_and_S/A_2/diann_out.log.txt\n",
      "Finished\n",
      "\n",
      "-------- Get quant for replicate 3 in experiment group Aposymb\n",
      "\n",
      "\n",
      "DIA-NN 1.7.12 (Data Independent Acquisition by Neural Networks)\n",
      "Compiled on Feb 13 2021 17:30:41\n",
      "Current date and time: Mon Mar  1 12:43:14 2021\n",
      "Logical CPU cores: 8\n",
      "\n",
      "1 files will be processed\n",
      "[0:00] Loading spectral library /home/ekvall/Desktop/sample_A_and_S/A_3/library.tsv\n",
      "[0:00] Finding proteotypic peptides (assuming that the list of UniProt ids provided for each peptide is complete)\n",
      "[0:00] Spectral library loaded: 170 protein isoforms, 170 protein groups and 185 precursors in 185 elution groups.\n",
      "[0:00] Initialising library\n",
      "[0:00] Saving the library to /home/ekvall/Desktop/sample_A_and_S/A_3/library.tsv.speclib\n",
      "\n",
      "[0:00] File #1/1\n",
      "[0:00] Loading run /sdd/proteomics/DATA/PXD014076/Aposymb/mzML/Aposymb_Proteome_DIA_RAW_A03.mzML\n",
      "[0:26] 185 library precursors are potentially detectable\n",
      "[0:26] Processing...\n",
      "[0:29] RT window set to 2.33343\n",
      "[0:29] Peak width: 4.57459\n",
      "[0:29] Scan window radius set to 10\n",
      "[0:30] Optimised mass accuracy: 4.28921 ppm\n",
      "[0:30] Removing interfering precursors\n",
      "[0:30] Training the neural network: 185 targets, 125 decoys\n",
      "[0:30] Number of IDs at 0.01 FDR: 177\n",
      "[0:30] Too low number of IDs with NNs: reverting to the linear classifier\n",
      "[0:30] Number of IDs at 0.01 FDR: 174\n",
      "[0:30] Calculating protein q-values\n",
      "[0:30] Number of protein isoforms identified at 1% FDR: 159 (precursor-level), 14 (protein-level) (inference performed using proteotypic peptides only)\n",
      "[0:30] Quantification\n",
      "[0:31] Quantification information saved to /sdd/proteomics/DATA/PXD014076/Aposymb/mzML/Aposymb_Proteome_DIA_RAW_A03.mzML.quant.\n",
      "\n",
      "[0:31] Cross-run analysis\n",
      "[0:31] Reading quantification information: 1 files\n",
      "[0:31] Quantifying peptides\n",
      "[0:31] Assembling protein groups\n",
      "[0:31] Quantifying proteins\n",
      "[0:31] Calculating q-values for protein and gene groups\n",
      "[0:31] Writing report\n",
      "[0:31] Report saved to /home/ekvall/Desktop/sample_A_and_S/A_3/diann_out.tsv.\n",
      "[0:31] Stats report saved to /home/ekvall/Desktop/sample_A_and_S/A_3/diann_out.stats.tsv\n",
      "[0:31] Writing gene report\n",
      "[0:31] Gene report saved to report.genes.tsv.\n",
      "[0:31] Log saved to /home/ekvall/Desktop/sample_A_and_S/A_3/diann_out.log.txt\n",
      "Finished\n",
      "\n",
      "-------- Get quant for replicate 4 in experiment group Aposymb\n",
      "\n",
      "\n",
      "DIA-NN 1.7.12 (Data Independent Acquisition by Neural Networks)\n",
      "Compiled on Feb 13 2021 17:30:41\n",
      "Current date and time: Mon Mar  1 12:43:45 2021\n",
      "Logical CPU cores: 8\n",
      "\n",
      "1 files will be processed\n",
      "[0:00] Loading spectral library /home/ekvall/Desktop/sample_A_and_S/A_4/library.tsv\n",
      "[0:00] Finding proteotypic peptides (assuming that the list of UniProt ids provided for each peptide is complete)\n",
      "[0:00] Spectral library loaded: 149 protein isoforms, 149 protein groups and 166 precursors in 166 elution groups.\n",
      "[0:00] Initialising library\n",
      "[0:00] Saving the library to /home/ekvall/Desktop/sample_A_and_S/A_4/library.tsv.speclib\n",
      "\n",
      "[0:00] File #1/1\n",
      "[0:00] Loading run /sdd/proteomics/DATA/PXD014076/Aposymb/mzML/Aposymb_Proteome_DIA_RAW_A04.mzML\n",
      "[0:25] 166 library precursors are potentially detectable\n",
      "[0:25] Processing...\n",
      "[0:28] RT window set to 2.37239\n",
      "[0:28] Peak width: 4.78261\n",
      "[0:28] Scan window radius set to 10\n",
      "[0:29] Optimised mass accuracy: 9.33558 ppm\n",
      "[0:29] Removing interfering precursors\n",
      "[0:29] Training the neural network: 165 targets, 135 decoys\n",
      "[0:29] Number of IDs at 0.01 FDR: 164\n",
      "[0:29] Calculating protein q-values\n",
      "[0:29] Number of protein isoforms identified at 1% FDR: 148 (precursor-level), 15 (protein-level) (inference performed using proteotypic peptides only)\n",
      "[0:29] Quantification\n",
      "[0:29] Quantification information saved to /sdd/proteomics/DATA/PXD014076/Aposymb/mzML/Aposymb_Proteome_DIA_RAW_A04.mzML.quant.\n",
      "\n",
      "[0:29] Cross-run analysis\n",
      "[0:29] Reading quantification information: 1 files\n",
      "[0:29] Quantifying peptides\n",
      "[0:29] Assembling protein groups\n",
      "[0:29] Quantifying proteins\n",
      "[0:29] Calculating q-values for protein and gene groups\n",
      "[0:29] Writing report\n",
      "[0:29] Report saved to /home/ekvall/Desktop/sample_A_and_S/A_4/diann_out.tsv.\n",
      "[0:29] Stats report saved to /home/ekvall/Desktop/sample_A_and_S/A_4/diann_out.stats.tsv\n",
      "[0:29] Writing gene report\n",
      "[0:29] Gene report saved to report.genes.tsv.\n",
      "[0:29] Log saved to /home/ekvall/Desktop/sample_A_and_S/A_4/diann_out.log.txt\n",
      "Finished\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------- Get quant for replicate 5 in experiment group Aposymb\n",
      "\n",
      "\n",
      "DIA-NN 1.7.12 (Data Independent Acquisition by Neural Networks)\n",
      "Compiled on Feb 13 2021 17:30:41\n",
      "Current date and time: Mon Mar  1 12:44:15 2021\n",
      "Logical CPU cores: 8\n",
      "\n",
      "1 files will be processed\n",
      "[0:00] Loading spectral library /home/ekvall/Desktop/sample_A_and_S/A_5/library.tsv\n",
      "[0:00] Finding proteotypic peptides (assuming that the list of UniProt ids provided for each peptide is complete)\n",
      "[0:00] Spectral library loaded: 149 protein isoforms, 149 protein groups and 161 precursors in 161 elution groups.\n",
      "[0:00] Initialising library\n",
      "[0:00] Saving the library to /home/ekvall/Desktop/sample_A_and_S/A_5/library.tsv.speclib\n",
      "\n",
      "[0:00] File #1/1\n",
      "[0:00] Loading run /sdd/proteomics/DATA/PXD014076/Aposymb/mzML/Aposymb_Proteome_DIA_RAW_A05.mzML\n",
      "[0:25] 161 library precursors are potentially detectable\n",
      "[0:25] Processing...\n",
      "[0:27] RT window set to 2.33863\n",
      "[0:27] Peak width: 4.78261\n",
      "[0:27] Scan window radius set to 10\n",
      "[0:29] Optimised mass accuracy: 6.16682 ppm\n",
      "[0:29] Removing interfering precursors\n",
      "[0:29] Training the neural network: 161 targets, 117 decoys\n",
      "[0:29] Number of IDs at 0.01 FDR: 161\n",
      "[0:29] Calculating protein q-values\n",
      "[0:29] Number of protein isoforms identified at 1% FDR: 149 (precursor-level), 0 (protein-level) (inference performed using proteotypic peptides only)\n",
      "[0:29] Quantification\n",
      "[0:29] Quantification information saved to /sdd/proteomics/DATA/PXD014076/Aposymb/mzML/Aposymb_Proteome_DIA_RAW_A05.mzML.quant.\n",
      "\n",
      "[0:29] Cross-run analysis\n",
      "[0:29] Reading quantification information: 1 files\n",
      "[0:29] Quantifying peptides\n",
      "[0:29] Assembling protein groups\n",
      "[0:29] Quantifying proteins\n",
      "[0:29] Calculating q-values for protein and gene groups\n",
      "[0:29] Writing report\n",
      "[0:29] Report saved to /home/ekvall/Desktop/sample_A_and_S/A_5/diann_out.tsv.\n",
      "[0:29] Stats report saved to /home/ekvall/Desktop/sample_A_and_S/A_5/diann_out.stats.tsv\n",
      "[0:29] Writing gene report\n",
      "[0:29] Gene report saved to report.genes.tsv.\n",
      "[0:29] Log saved to /home/ekvall/Desktop/sample_A_and_S/A_5/diann_out.log.txt\n",
      "Finished\n",
      "\n"
     ]
    }
   ],
   "source": [
    "runDiann(\"A\",\"Aposymb\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Run for each replicate in the Symb group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------- Get quant for replicate 1 in experiment group Symb\n",
      "\n",
      "\n",
      "DIA-NN 1.7.12 (Data Independent Acquisition by Neural Networks)\n",
      "Compiled on Feb 13 2021 17:30:41\n",
      "Current date and time: Mon Mar  1 12:44:47 2021\n",
      "Logical CPU cores: 8\n",
      "\n",
      "1 files will be processed\n",
      "[0:00] Loading spectral library /home/ekvall/Desktop/sample_A_and_S/S_1/library.tsv\n",
      "[0:00] Finding proteotypic peptides (assuming that the list of UniProt ids provided for each peptide is complete)\n",
      "[0:00] Spectral library loaded: 164 protein isoforms, 164 protein groups and 184 precursors in 184 elution groups.\n",
      "[0:00] Initialising library\n",
      "[0:00] Saving the library to /home/ekvall/Desktop/sample_A_and_S/S_1/library.tsv.speclib\n",
      "\n",
      "[0:00] File #1/1\n",
      "[0:00] Loading run /sdd/proteomics/DATA/PXD014076/Symb/mzML/Symb_Proteome_DIA_RAW_S01.mzML\n",
      "[0:26] 184 library precursors are potentially detectable\n",
      "[0:26] Processing...\n",
      "[0:29] RT window set to 2.40145\n",
      "[0:29] Peak width: 4.79006\n",
      "[0:29] Scan window radius set to 10\n",
      "[0:31] Optimised mass accuracy: 7.3019 ppm\n",
      "[0:31] Removing interfering precursors\n",
      "[0:31] Training the neural network: 184 targets, 130 decoys\n",
      "[0:31] Number of IDs at 0.01 FDR: 179\n",
      "[0:31] Calculating protein q-values\n",
      "[0:31] Number of protein isoforms identified at 1% FDR: 160 (precursor-level), 0 (protein-level) (inference performed using proteotypic peptides only)\n",
      "[0:31] Quantification\n",
      "[0:31] Quantification information saved to /sdd/proteomics/DATA/PXD014076/Symb/mzML/Symb_Proteome_DIA_RAW_S01.mzML.quant.\n",
      "\n",
      "[0:31] Cross-run analysis\n",
      "[0:31] Reading quantification information: 1 files\n",
      "[0:31] Quantifying peptides\n",
      "[0:31] Assembling protein groups\n",
      "[0:31] Quantifying proteins\n",
      "[0:31] Calculating q-values for protein and gene groups\n",
      "[0:31] Writing report\n",
      "[0:31] Report saved to /home/ekvall/Desktop/sample_A_and_S/S_1/diann_out.tsv.\n",
      "[0:31] Stats report saved to /home/ekvall/Desktop/sample_A_and_S/S_1/diann_out.stats.tsv\n",
      "[0:31] Writing gene report\n",
      "[0:31] Gene report saved to report.genes.tsv.\n",
      "[0:31] Log saved to /home/ekvall/Desktop/sample_A_and_S/S_1/diann_out.log.txt\n",
      "Finished\n",
      "\n",
      "-------- Get quant for replicate 2 in experiment group Symb\n",
      "\n",
      "\n",
      "DIA-NN 1.7.12 (Data Independent Acquisition by Neural Networks)\n",
      "Compiled on Feb 13 2021 17:30:41\n",
      "Current date and time: Mon Mar  1 12:45:18 2021\n",
      "Logical CPU cores: 8\n",
      "\n",
      "1 files will be processed\n",
      "[0:00] Loading spectral library /home/ekvall/Desktop/sample_A_and_S/S_2/library.tsv\n",
      "[0:00] Finding proteotypic peptides (assuming that the list of UniProt ids provided for each peptide is complete)\n",
      "[0:00] Spectral library loaded: 165 protein isoforms, 165 protein groups and 186 precursors in 186 elution groups.\n",
      "[0:00] Initialising library\n",
      "[0:00] Saving the library to /home/ekvall/Desktop/sample_A_and_S/S_2/library.tsv.speclib\n",
      "\n",
      "[0:00] File #1/1\n",
      "[0:00] Loading run /sdd/proteomics/DATA/PXD014076/Symb/mzML/Symb_Proteome_DIA_RAW_S02.mzML\n",
      "[0:26] 186 library precursors are potentially detectable\n",
      "[0:26] Processing...\n",
      "[0:29] RT window set to 2.37119\n",
      "[0:29] Peak width: 4.90811\n",
      "[0:29] Scan window radius set to 10\n",
      "[0:30] Optimised mass accuracy: 8.75304 ppm\n",
      "[0:31] Removing interfering precursors\n",
      "[0:31] Training the neural network: 186 targets, 136 decoys\n",
      "[0:31] Number of IDs at 0.01 FDR: 186\n",
      "[0:31] Calculating protein q-values\n",
      "[0:31] Number of protein isoforms identified at 1% FDR: 165 (precursor-level), 0 (protein-level) (inference performed using proteotypic peptides only)\n",
      "[0:31] Quantification\n",
      "[0:31] Quantification information saved to /sdd/proteomics/DATA/PXD014076/Symb/mzML/Symb_Proteome_DIA_RAW_S02.mzML.quant.\n",
      "\n",
      "[0:31] Cross-run analysis\n",
      "[0:31] Reading quantification information: 1 files\n",
      "[0:31] Quantifying peptides\n",
      "[0:31] Assembling protein groups\n",
      "[0:31] Quantifying proteins\n",
      "[0:31] Calculating q-values for protein and gene groups\n",
      "[0:31] Writing report\n",
      "[0:31] Report saved to /home/ekvall/Desktop/sample_A_and_S/S_2/diann_out.tsv.\n",
      "[0:31] Stats report saved to /home/ekvall/Desktop/sample_A_and_S/S_2/diann_out.stats.tsv\n",
      "[0:31] Writing gene report\n",
      "[0:31] Gene report saved to report.genes.tsv.\n",
      "[0:31] Log saved to /home/ekvall/Desktop/sample_A_and_S/S_2/diann_out.log.txt\n",
      "Finished\n",
      "\n",
      "-------- Get quant for replicate 3 in experiment group Symb\n",
      "\n",
      "\n",
      "DIA-NN 1.7.12 (Data Independent Acquisition by Neural Networks)\n",
      "Compiled on Feb 13 2021 17:30:41\n",
      "Current date and time: Mon Mar  1 12:45:50 2021\n",
      "Logical CPU cores: 8\n",
      "\n",
      "1 files will be processed\n",
      "[0:00] Loading spectral library /home/ekvall/Desktop/sample_A_and_S/S_3/library.tsv\n",
      "[0:00] Finding proteotypic peptides (assuming that the list of UniProt ids provided for each peptide is complete)\n",
      "[0:00] Spectral library loaded: 161 protein isoforms, 161 protein groups and 181 precursors in 181 elution groups.\n",
      "[0:00] Initialising library\n",
      "[0:00] Saving the library to /home/ekvall/Desktop/sample_A_and_S/S_3/library.tsv.speclib\n",
      "\n",
      "[0:00] File #1/1\n",
      "[0:00] Loading run /sdd/proteomics/DATA/PXD014076/Symb/mzML/Symb_Proteome_DIA_RAW_S03.mzML\n",
      "[0:26] 181 library precursors are potentially detectable\n",
      "[0:26] Processing...\n",
      "[0:28] RT window set to 2.3702\n",
      "[0:28] Peak width: 4.88764\n",
      "[0:28] Scan window radius set to 10\n",
      "[0:30] Optimised mass accuracy: 6.24785 ppm\n",
      "[0:30] Removing interfering precursors\n",
      "[0:30] Training the neural network: 180 targets, 129 decoys\n",
      "[0:30] Number of IDs at 0.01 FDR: 177\n",
      "[0:30] Calculating protein q-values\n",
      "[0:30] Number of protein isoforms identified at 1% FDR: 158 (precursor-level), 17 (protein-level) (inference performed using proteotypic peptides only)\n",
      "[0:30] Quantification\n",
      "[0:30] Quantification information saved to /sdd/proteomics/DATA/PXD014076/Symb/mzML/Symb_Proteome_DIA_RAW_S03.mzML.quant.\n",
      "\n",
      "[0:30] Cross-run analysis\n",
      "[0:30] Reading quantification information: 1 files\n",
      "[0:30] Quantifying peptides\n",
      "[0:30] Assembling protein groups\n",
      "[0:30] Quantifying proteins\n",
      "[0:30] Calculating q-values for protein and gene groups\n",
      "[0:30] Writing report\n",
      "[0:30] Report saved to /home/ekvall/Desktop/sample_A_and_S/S_3/diann_out.tsv.\n",
      "[0:30] Stats report saved to /home/ekvall/Desktop/sample_A_and_S/S_3/diann_out.stats.tsv\n",
      "[0:30] Writing gene report\n",
      "[0:30] Gene report saved to report.genes.tsv.\n",
      "[0:30] Log saved to /home/ekvall/Desktop/sample_A_and_S/S_3/diann_out.log.txt\n",
      "Finished\n",
      "\n",
      "-------- Get quant for replicate 4 in experiment group Symb\n",
      "\n",
      "\n",
      "DIA-NN 1.7.12 (Data Independent Acquisition by Neural Networks)\n",
      "Compiled on Feb 13 2021 17:30:41\n",
      "Current date and time: Mon Mar  1 12:46:21 2021\n",
      "Logical CPU cores: 8\n",
      "\n",
      "1 files will be processed\n",
      "[0:00] Loading spectral library /home/ekvall/Desktop/sample_A_and_S/S_4/library.tsv\n",
      "[0:00] Finding proteotypic peptides (assuming that the list of UniProt ids provided for each peptide is complete)\n",
      "[0:00] Spectral library loaded: 131 protein isoforms, 131 protein groups and 146 precursors in 146 elution groups.\n",
      "[0:00] Initialising library\n",
      "[0:00] Saving the library to /home/ekvall/Desktop/sample_A_and_S/S_4/library.tsv.speclib\n",
      "\n",
      "[0:00] File #1/1\n",
      "[0:00] Loading run /sdd/proteomics/DATA/PXD014076/Symb/mzML/Symb_Proteome_DIA_RAW_S04.mzML\n",
      "[0:24] 146 library precursors are potentially detectable\n",
      "[0:24] Processing...\n",
      "[0:26] RT window set to 2.36992\n",
      "[0:26] Peak width: 4.73103\n",
      "[0:26] Scan window radius set to 10\n",
      "[0:27] Optimised mass accuracy: 4.35202 ppm\n",
      "[0:27] Removing interfering precursors\n",
      "[0:27] Too few confident identifications, neural network will not be used\n",
      "[0:27] Number of IDs at 0.01 FDR: 122\n",
      "[0:27] Number of IDs at 0.01 FDR: 136\n",
      "[0:27] Calculating protein q-values\n",
      "[0:27] Number of protein isoforms identified at 1% FDR: 123 (precursor-level), 11 (protein-level) (inference performed using proteotypic peptides only)\n",
      "[0:27] Quantification\n",
      "[0:27] Quantification information saved to /sdd/proteomics/DATA/PXD014076/Symb/mzML/Symb_Proteome_DIA_RAW_S04.mzML.quant.\n",
      "\n",
      "[0:27] Cross-run analysis\n",
      "[0:27] Reading quantification information: 1 files\n",
      "[0:27] Quantifying peptides\n",
      "[0:27] Assembling protein groups\n",
      "[0:27] Quantifying proteins\n",
      "[0:27] Calculating q-values for protein and gene groups\n",
      "[0:27] Writing report\n",
      "[0:27] Report saved to /home/ekvall/Desktop/sample_A_and_S/S_4/diann_out.tsv.\n",
      "[0:27] Stats report saved to /home/ekvall/Desktop/sample_A_and_S/S_4/diann_out.stats.tsv\n",
      "[0:27] Writing gene report\n",
      "[0:27] Gene report saved to report.genes.tsv.\n",
      "[0:27] Log saved to /home/ekvall/Desktop/sample_A_and_S/S_4/diann_out.log.txt\n",
      "Finished\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------- Get quant for replicate 5 in experiment group Symb\n",
      "\n",
      "\n",
      "DIA-NN 1.7.12 (Data Independent Acquisition by Neural Networks)\n",
      "Compiled on Feb 13 2021 17:30:41\n",
      "Current date and time: Mon Mar  1 12:46:49 2021\n",
      "Logical CPU cores: 8\n",
      "\n",
      "1 files will be processed\n",
      "[0:00] Loading spectral library /home/ekvall/Desktop/sample_A_and_S/S_5/library.tsv\n",
      "[0:00] Finding proteotypic peptides (assuming that the list of UniProt ids provided for each peptide is complete)\n",
      "[0:00] Spectral library loaded: 154 protein isoforms, 154 protein groups and 169 precursors in 169 elution groups.\n",
      "[0:00] Initialising library\n",
      "[0:00] Saving the library to /home/ekvall/Desktop/sample_A_and_S/S_5/library.tsv.speclib\n",
      "\n",
      "[0:00] File #1/1\n",
      "[0:00] Loading run /sdd/proteomics/DATA/PXD014076/Symb/mzML/Symb_Proteome_DIA_RAW_S05.mzML\n",
      "[0:24] 169 library precursors are potentially detectable\n",
      "[0:24] Processing...\n",
      "[0:26] RT window set to 2.38424\n",
      "[0:26] Peak width: 4.69048\n",
      "[0:26] Scan window radius set to 10\n",
      "[0:28] Optimised mass accuracy: 8.86491 ppm\n",
      "[0:28] Removing interfering precursors\n",
      "[0:28] Training the neural network: 169 targets, 124 decoys\n",
      "[0:28] Number of IDs at 0.01 FDR: 167\n",
      "[0:28] Too low number of IDs with NNs: reverting to the linear classifier\n",
      "[0:28] Number of IDs at 0.01 FDR: 169\n",
      "[0:28] Calculating protein q-values\n",
      "[0:28] Number of protein isoforms identified at 1% FDR: 154 (precursor-level), 0 (protein-level) (inference performed using proteotypic peptides only)\n",
      "[0:28] Quantification\n",
      "[0:28] Quantification information saved to /sdd/proteomics/DATA/PXD014076/Symb/mzML/Symb_Proteome_DIA_RAW_S05.mzML.quant.\n",
      "\n",
      "[0:28] Cross-run analysis\n",
      "[0:28] Reading quantification information: 1 files\n",
      "[0:28] Quantifying peptides\n",
      "[0:28] Assembling protein groups\n",
      "[0:28] Quantifying proteins\n",
      "[0:28] Calculating q-values for protein and gene groups\n",
      "[0:28] Writing report\n",
      "[0:28] Report saved to /home/ekvall/Desktop/sample_A_and_S/S_5/diann_out.tsv.\n",
      "[0:28] Stats report saved to /home/ekvall/Desktop/sample_A_and_S/S_5/diann_out.stats.tsv\n",
      "[0:28] Writing gene report\n",
      "[0:28] Gene report saved to report.genes.tsv.\n",
      "[0:28] Log saved to /home/ekvall/Desktop/sample_A_and_S/S_5/diann_out.log.txt\n",
      "Finished\n",
      "\n"
     ]
    }
   ],
   "source": [
    "runDiann(\"S\",\"Symb\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Collect all diann output for each replicate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {},
   "outputs": [],
   "source": [
    "def getQuantFram(short_name_list):\n",
    "    data = list()\n",
    "    for short_name in short_name_list:\n",
    "        for r_n in range(1,6):\n",
    "            path = '/home/ekvall/Desktop/sample_A_and_S/{}_{}'.format(short_name, r_n) + \"/library.tsv\"\n",
    "            \n",
    "            df = pd.read_csv(path, sep=\"\\t\")\n",
    "            if short_name == \"A\":\n",
    "                \n",
    "                df[\"Run\"] = [\"Aposymb_Proteome_DIA_RAW_A0{}\".format(r_n)] * df.shape[0]\n",
    "                \n",
    "            else:\n",
    "                df[\"Run\"] = [\"Symb_Proteome_DIA_RAW_S0{}\".format(r_n)] * df.shape[0]\n",
    "            data.append(df)\n",
    "    return pd.concat(data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_one = getQuantFram([\"A\", \"S\"])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['PrecursorMz', 'ProductMz', 'Annotation', 'ProteinId', 'GeneName',\n",
       "       'PeptideSequence', 'ModifiedPeptideSequence', 'PrecursorCharge',\n",
       "       'LibraryIntensity', 'NormalizedRetentionTime', 'PrecursorIonMobility',\n",
       "       'FragmentType', 'FragmentCharge', 'FragmentSeriesNumber',\n",
       "       'FragmentLossType', 'Run'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 157,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_one.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 156,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>1046</th>\n",
       "      <th>1047</th>\n",
       "      <th>1048</th>\n",
       "      <th>1049</th>\n",
       "      <th>1050</th>\n",
       "      <th>1051</th>\n",
       "      <th>1052</th>\n",
       "      <th>1053</th>\n",
       "      <th>1054</th>\n",
       "      <th>967</th>\n",
       "      <th>...</th>\n",
       "      <th>978</th>\n",
       "      <th>979</th>\n",
       "      <th>980</th>\n",
       "      <th>981</th>\n",
       "      <th>982</th>\n",
       "      <th>983</th>\n",
       "      <th>984</th>\n",
       "      <th>985</th>\n",
       "      <th>986</th>\n",
       "      <th>987</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>PrecursorMz</th>\n",
       "      <td>738.038</td>\n",
       "      <td>738.038</td>\n",
       "      <td>738.038</td>\n",
       "      <td>738.038</td>\n",
       "      <td>738.038</td>\n",
       "      <td>738.038</td>\n",
       "      <td>738.038</td>\n",
       "      <td>738.038</td>\n",
       "      <td>738.038</td>\n",
       "      <td>738.038</td>\n",
       "      <td>...</td>\n",
       "      <td>738.038</td>\n",
       "      <td>738.038</td>\n",
       "      <td>738.038</td>\n",
       "      <td>738.038</td>\n",
       "      <td>738.038</td>\n",
       "      <td>738.038</td>\n",
       "      <td>738.038</td>\n",
       "      <td>738.038</td>\n",
       "      <td>738.038</td>\n",
       "      <td>738.038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ProductMz</th>\n",
       "      <td>789.399</td>\n",
       "      <td>854.425</td>\n",
       "      <td>902.483</td>\n",
       "      <td>967.509</td>\n",
       "      <td>1081.55</td>\n",
       "      <td>1131.55</td>\n",
       "      <td>1195.6</td>\n",
       "      <td>1245.6</td>\n",
       "      <td>1310.62</td>\n",
       "      <td>199.108</td>\n",
       "      <td>...</td>\n",
       "      <td>725.383</td>\n",
       "      <td>789.399</td>\n",
       "      <td>854.425</td>\n",
       "      <td>902.483</td>\n",
       "      <td>967.509</td>\n",
       "      <td>1081.55</td>\n",
       "      <td>1131.55</td>\n",
       "      <td>1195.6</td>\n",
       "      <td>1245.6</td>\n",
       "      <td>1310.62</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Annotation</th>\n",
       "      <td>b7^1</td>\n",
       "      <td>y7^1</td>\n",
       "      <td>b8^1</td>\n",
       "      <td>y8^1</td>\n",
       "      <td>y9^1</td>\n",
       "      <td>b10^1</td>\n",
       "      <td>y10^1</td>\n",
       "      <td>b11^1</td>\n",
       "      <td>y11^1</td>\n",
       "      <td>y3^2</td>\n",
       "      <td>...</td>\n",
       "      <td>y6^1</td>\n",
       "      <td>b7^1</td>\n",
       "      <td>y7^1</td>\n",
       "      <td>b8^1</td>\n",
       "      <td>y8^1</td>\n",
       "      <td>y9^1</td>\n",
       "      <td>b10^1</td>\n",
       "      <td>y10^1</td>\n",
       "      <td>b11^1</td>\n",
       "      <td>y11^1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ProteinId</th>\n",
       "      <td>P13645</td>\n",
       "      <td>P13645</td>\n",
       "      <td>P13645</td>\n",
       "      <td>P13645</td>\n",
       "      <td>P13645</td>\n",
       "      <td>P13645</td>\n",
       "      <td>P13645</td>\n",
       "      <td>P13645</td>\n",
       "      <td>P13645</td>\n",
       "      <td>P13645</td>\n",
       "      <td>...</td>\n",
       "      <td>P13645</td>\n",
       "      <td>P13645</td>\n",
       "      <td>P13645</td>\n",
       "      <td>P13645</td>\n",
       "      <td>P13645</td>\n",
       "      <td>P13645</td>\n",
       "      <td>P13645</td>\n",
       "      <td>P13645</td>\n",
       "      <td>P13645</td>\n",
       "      <td>P13645</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GeneName</th>\n",
       "      <td>KRT10</td>\n",
       "      <td>KRT10</td>\n",
       "      <td>KRT10</td>\n",
       "      <td>KRT10</td>\n",
       "      <td>KRT10</td>\n",
       "      <td>KRT10</td>\n",
       "      <td>KRT10</td>\n",
       "      <td>KRT10</td>\n",
       "      <td>KRT10</td>\n",
       "      <td>KRT10</td>\n",
       "      <td>...</td>\n",
       "      <td>KRT10</td>\n",
       "      <td>KRT10</td>\n",
       "      <td>KRT10</td>\n",
       "      <td>KRT10</td>\n",
       "      <td>KRT10</td>\n",
       "      <td>KRT10</td>\n",
       "      <td>KRT10</td>\n",
       "      <td>KRT10</td>\n",
       "      <td>KRT10</td>\n",
       "      <td>KRT10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PeptideSequence</th>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "      <td>...</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ModifiedPeptideSequence</th>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "      <td>...</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PrecursorCharge</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LibraryIntensity</th>\n",
       "      <td>2149.52</td>\n",
       "      <td>10000</td>\n",
       "      <td>1161.01</td>\n",
       "      <td>2303.11</td>\n",
       "      <td>1563.38</td>\n",
       "      <td>984.853</td>\n",
       "      <td>456.487</td>\n",
       "      <td>1872.84</td>\n",
       "      <td>980.877</td>\n",
       "      <td>1508.02</td>\n",
       "      <td>...</td>\n",
       "      <td>3358.87</td>\n",
       "      <td>3024.49</td>\n",
       "      <td>10000</td>\n",
       "      <td>1396.32</td>\n",
       "      <td>4397.39</td>\n",
       "      <td>2479.5</td>\n",
       "      <td>1516.18</td>\n",
       "      <td>1328.61</td>\n",
       "      <td>1989.08</td>\n",
       "      <td>499.282</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NormalizedRetentionTime</th>\n",
       "      <td>83.4312</td>\n",
       "      <td>83.4312</td>\n",
       "      <td>83.4312</td>\n",
       "      <td>83.4312</td>\n",
       "      <td>83.4312</td>\n",
       "      <td>83.4312</td>\n",
       "      <td>83.4312</td>\n",
       "      <td>83.4312</td>\n",
       "      <td>83.4312</td>\n",
       "      <td>83.1712</td>\n",
       "      <td>...</td>\n",
       "      <td>83.1712</td>\n",
       "      <td>83.1712</td>\n",
       "      <td>83.1712</td>\n",
       "      <td>83.1712</td>\n",
       "      <td>83.1712</td>\n",
       "      <td>83.1712</td>\n",
       "      <td>83.1712</td>\n",
       "      <td>83.1712</td>\n",
       "      <td>83.1712</td>\n",
       "      <td>83.1712</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PrecursorIonMobility</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FragmentType</th>\n",
       "      <td>b</td>\n",
       "      <td>y</td>\n",
       "      <td>b</td>\n",
       "      <td>y</td>\n",
       "      <td>y</td>\n",
       "      <td>b</td>\n",
       "      <td>y</td>\n",
       "      <td>b</td>\n",
       "      <td>y</td>\n",
       "      <td>y</td>\n",
       "      <td>...</td>\n",
       "      <td>y</td>\n",
       "      <td>b</td>\n",
       "      <td>y</td>\n",
       "      <td>b</td>\n",
       "      <td>y</td>\n",
       "      <td>y</td>\n",
       "      <td>b</td>\n",
       "      <td>y</td>\n",
       "      <td>b</td>\n",
       "      <td>y</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FragmentCharge</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FragmentSeriesNumber</th>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>...</td>\n",
       "      <td>6</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FragmentLossType</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Run</th>\n",
       "      <td>Symb_Proteome_DIA_RAW_S03</td>\n",
       "      <td>Symb_Proteome_DIA_RAW_S03</td>\n",
       "      <td>Symb_Proteome_DIA_RAW_S03</td>\n",
       "      <td>Symb_Proteome_DIA_RAW_S03</td>\n",
       "      <td>Symb_Proteome_DIA_RAW_S03</td>\n",
       "      <td>Symb_Proteome_DIA_RAW_S03</td>\n",
       "      <td>Symb_Proteome_DIA_RAW_S03</td>\n",
       "      <td>Symb_Proteome_DIA_RAW_S03</td>\n",
       "      <td>Symb_Proteome_DIA_RAW_S03</td>\n",
       "      <td>Symb_Proteome_DIA_RAW_S05</td>\n",
       "      <td>...</td>\n",
       "      <td>Symb_Proteome_DIA_RAW_S05</td>\n",
       "      <td>Symb_Proteome_DIA_RAW_S05</td>\n",
       "      <td>Symb_Proteome_DIA_RAW_S05</td>\n",
       "      <td>Symb_Proteome_DIA_RAW_S05</td>\n",
       "      <td>Symb_Proteome_DIA_RAW_S05</td>\n",
       "      <td>Symb_Proteome_DIA_RAW_S05</td>\n",
       "      <td>Symb_Proteome_DIA_RAW_S05</td>\n",
       "      <td>Symb_Proteome_DIA_RAW_S05</td>\n",
       "      <td>Symb_Proteome_DIA_RAW_S05</td>\n",
       "      <td>Symb_Proteome_DIA_RAW_S05</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>16 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                              1046                       1047  \\\n",
       "PrecursorMz                                738.038                    738.038   \n",
       "ProductMz                                  789.399                    854.425   \n",
       "Annotation                                    b7^1                       y7^1   \n",
       "ProteinId                                   P13645                     P13645   \n",
       "GeneName                                     KRT10                      KRT10   \n",
       "PeptideSequence                SKELTTEIDNNIEQISSYK        SKELTTEIDNNIEQISSYK   \n",
       "ModifiedPeptideSequence        SKELTTEIDNNIEQISSYK        SKELTTEIDNNIEQISSYK   \n",
       "PrecursorCharge                                  3                          3   \n",
       "LibraryIntensity                           2149.52                      10000   \n",
       "NormalizedRetentionTime                    83.4312                    83.4312   \n",
       "PrecursorIonMobility                           NaN                        NaN   \n",
       "FragmentType                                     b                          y   \n",
       "FragmentCharge                                   1                          1   \n",
       "FragmentSeriesNumber                             7                          7   \n",
       "FragmentLossType                               NaN                        NaN   \n",
       "Run                      Symb_Proteome_DIA_RAW_S03  Symb_Proteome_DIA_RAW_S03   \n",
       "\n",
       "                                              1048                       1049  \\\n",
       "PrecursorMz                                738.038                    738.038   \n",
       "ProductMz                                  902.483                    967.509   \n",
       "Annotation                                    b8^1                       y8^1   \n",
       "ProteinId                                   P13645                     P13645   \n",
       "GeneName                                     KRT10                      KRT10   \n",
       "PeptideSequence                SKELTTEIDNNIEQISSYK        SKELTTEIDNNIEQISSYK   \n",
       "ModifiedPeptideSequence        SKELTTEIDNNIEQISSYK        SKELTTEIDNNIEQISSYK   \n",
       "PrecursorCharge                                  3                          3   \n",
       "LibraryIntensity                           1161.01                    2303.11   \n",
       "NormalizedRetentionTime                    83.4312                    83.4312   \n",
       "PrecursorIonMobility                           NaN                        NaN   \n",
       "FragmentType                                     b                          y   \n",
       "FragmentCharge                                   1                          1   \n",
       "FragmentSeriesNumber                             8                          8   \n",
       "FragmentLossType                               NaN                        NaN   \n",
       "Run                      Symb_Proteome_DIA_RAW_S03  Symb_Proteome_DIA_RAW_S03   \n",
       "\n",
       "                                              1050                       1051  \\\n",
       "PrecursorMz                                738.038                    738.038   \n",
       "ProductMz                                  1081.55                    1131.55   \n",
       "Annotation                                    y9^1                      b10^1   \n",
       "ProteinId                                   P13645                     P13645   \n",
       "GeneName                                     KRT10                      KRT10   \n",
       "PeptideSequence                SKELTTEIDNNIEQISSYK        SKELTTEIDNNIEQISSYK   \n",
       "ModifiedPeptideSequence        SKELTTEIDNNIEQISSYK        SKELTTEIDNNIEQISSYK   \n",
       "PrecursorCharge                                  3                          3   \n",
       "LibraryIntensity                           1563.38                    984.853   \n",
       "NormalizedRetentionTime                    83.4312                    83.4312   \n",
       "PrecursorIonMobility                           NaN                        NaN   \n",
       "FragmentType                                     y                          b   \n",
       "FragmentCharge                                   1                          1   \n",
       "FragmentSeriesNumber                             9                          0   \n",
       "FragmentLossType                               NaN                        NaN   \n",
       "Run                      Symb_Proteome_DIA_RAW_S03  Symb_Proteome_DIA_RAW_S03   \n",
       "\n",
       "                                              1052                       1053  \\\n",
       "PrecursorMz                                738.038                    738.038   \n",
       "ProductMz                                   1195.6                     1245.6   \n",
       "Annotation                                   y10^1                      b11^1   \n",
       "ProteinId                                   P13645                     P13645   \n",
       "GeneName                                     KRT10                      KRT10   \n",
       "PeptideSequence                SKELTTEIDNNIEQISSYK        SKELTTEIDNNIEQISSYK   \n",
       "ModifiedPeptideSequence        SKELTTEIDNNIEQISSYK        SKELTTEIDNNIEQISSYK   \n",
       "PrecursorCharge                                  3                          3   \n",
       "LibraryIntensity                           456.487                    1872.84   \n",
       "NormalizedRetentionTime                    83.4312                    83.4312   \n",
       "PrecursorIonMobility                           NaN                        NaN   \n",
       "FragmentType                                     y                          b   \n",
       "FragmentCharge                                   1                          1   \n",
       "FragmentSeriesNumber                             0                          1   \n",
       "FragmentLossType                               NaN                        NaN   \n",
       "Run                      Symb_Proteome_DIA_RAW_S03  Symb_Proteome_DIA_RAW_S03   \n",
       "\n",
       "                                              1054                       967   \\\n",
       "PrecursorMz                                738.038                    738.038   \n",
       "ProductMz                                  1310.62                    199.108   \n",
       "Annotation                                   y11^1                       y3^2   \n",
       "ProteinId                                   P13645                     P13645   \n",
       "GeneName                                     KRT10                      KRT10   \n",
       "PeptideSequence                SKELTTEIDNNIEQISSYK        SKELTTEIDNNIEQISSYK   \n",
       "ModifiedPeptideSequence        SKELTTEIDNNIEQISSYK        SKELTTEIDNNIEQISSYK   \n",
       "PrecursorCharge                                  3                          3   \n",
       "LibraryIntensity                           980.877                    1508.02   \n",
       "NormalizedRetentionTime                    83.4312                    83.1712   \n",
       "PrecursorIonMobility                           NaN                        NaN   \n",
       "FragmentType                                     y                          y   \n",
       "FragmentCharge                                   1                          2   \n",
       "FragmentSeriesNumber                             1                          3   \n",
       "FragmentLossType                               NaN                        NaN   \n",
       "Run                      Symb_Proteome_DIA_RAW_S03  Symb_Proteome_DIA_RAW_S05   \n",
       "\n",
       "                         ...                       978   \\\n",
       "PrecursorMz              ...                    738.038   \n",
       "ProductMz                ...                    725.383   \n",
       "Annotation               ...                       y6^1   \n",
       "ProteinId                ...                     P13645   \n",
       "GeneName                 ...                      KRT10   \n",
       "PeptideSequence          ...        SKELTTEIDNNIEQISSYK   \n",
       "ModifiedPeptideSequence  ...        SKELTTEIDNNIEQISSYK   \n",
       "PrecursorCharge          ...                          3   \n",
       "LibraryIntensity         ...                    3358.87   \n",
       "NormalizedRetentionTime  ...                    83.1712   \n",
       "PrecursorIonMobility     ...                        NaN   \n",
       "FragmentType             ...                          y   \n",
       "FragmentCharge           ...                          1   \n",
       "FragmentSeriesNumber     ...                          6   \n",
       "FragmentLossType         ...                        NaN   \n",
       "Run                      ...  Symb_Proteome_DIA_RAW_S05   \n",
       "\n",
       "                                              979                        980   \\\n",
       "PrecursorMz                                738.038                    738.038   \n",
       "ProductMz                                  789.399                    854.425   \n",
       "Annotation                                    b7^1                       y7^1   \n",
       "ProteinId                                   P13645                     P13645   \n",
       "GeneName                                     KRT10                      KRT10   \n",
       "PeptideSequence                SKELTTEIDNNIEQISSYK        SKELTTEIDNNIEQISSYK   \n",
       "ModifiedPeptideSequence        SKELTTEIDNNIEQISSYK        SKELTTEIDNNIEQISSYK   \n",
       "PrecursorCharge                                  3                          3   \n",
       "LibraryIntensity                           3024.49                      10000   \n",
       "NormalizedRetentionTime                    83.1712                    83.1712   \n",
       "PrecursorIonMobility                           NaN                        NaN   \n",
       "FragmentType                                     b                          y   \n",
       "FragmentCharge                                   1                          1   \n",
       "FragmentSeriesNumber                             7                          7   \n",
       "FragmentLossType                               NaN                        NaN   \n",
       "Run                      Symb_Proteome_DIA_RAW_S05  Symb_Proteome_DIA_RAW_S05   \n",
       "\n",
       "                                              981                        982   \\\n",
       "PrecursorMz                                738.038                    738.038   \n",
       "ProductMz                                  902.483                    967.509   \n",
       "Annotation                                    b8^1                       y8^1   \n",
       "ProteinId                                   P13645                     P13645   \n",
       "GeneName                                     KRT10                      KRT10   \n",
       "PeptideSequence                SKELTTEIDNNIEQISSYK        SKELTTEIDNNIEQISSYK   \n",
       "ModifiedPeptideSequence        SKELTTEIDNNIEQISSYK        SKELTTEIDNNIEQISSYK   \n",
       "PrecursorCharge                                  3                          3   \n",
       "LibraryIntensity                           1396.32                    4397.39   \n",
       "NormalizedRetentionTime                    83.1712                    83.1712   \n",
       "PrecursorIonMobility                           NaN                        NaN   \n",
       "FragmentType                                     b                          y   \n",
       "FragmentCharge                                   1                          1   \n",
       "FragmentSeriesNumber                             8                          8   \n",
       "FragmentLossType                               NaN                        NaN   \n",
       "Run                      Symb_Proteome_DIA_RAW_S05  Symb_Proteome_DIA_RAW_S05   \n",
       "\n",
       "                                              983                        984   \\\n",
       "PrecursorMz                                738.038                    738.038   \n",
       "ProductMz                                  1081.55                    1131.55   \n",
       "Annotation                                    y9^1                      b10^1   \n",
       "ProteinId                                   P13645                     P13645   \n",
       "GeneName                                     KRT10                      KRT10   \n",
       "PeptideSequence                SKELTTEIDNNIEQISSYK        SKELTTEIDNNIEQISSYK   \n",
       "ModifiedPeptideSequence        SKELTTEIDNNIEQISSYK        SKELTTEIDNNIEQISSYK   \n",
       "PrecursorCharge                                  3                          3   \n",
       "LibraryIntensity                            2479.5                    1516.18   \n",
       "NormalizedRetentionTime                    83.1712                    83.1712   \n",
       "PrecursorIonMobility                           NaN                        NaN   \n",
       "FragmentType                                     y                          b   \n",
       "FragmentCharge                                   1                          1   \n",
       "FragmentSeriesNumber                             9                          0   \n",
       "FragmentLossType                               NaN                        NaN   \n",
       "Run                      Symb_Proteome_DIA_RAW_S05  Symb_Proteome_DIA_RAW_S05   \n",
       "\n",
       "                                              985                        986   \\\n",
       "PrecursorMz                                738.038                    738.038   \n",
       "ProductMz                                   1195.6                     1245.6   \n",
       "Annotation                                   y10^1                      b11^1   \n",
       "ProteinId                                   P13645                     P13645   \n",
       "GeneName                                     KRT10                      KRT10   \n",
       "PeptideSequence                SKELTTEIDNNIEQISSYK        SKELTTEIDNNIEQISSYK   \n",
       "ModifiedPeptideSequence        SKELTTEIDNNIEQISSYK        SKELTTEIDNNIEQISSYK   \n",
       "PrecursorCharge                                  3                          3   \n",
       "LibraryIntensity                           1328.61                    1989.08   \n",
       "NormalizedRetentionTime                    83.1712                    83.1712   \n",
       "PrecursorIonMobility                           NaN                        NaN   \n",
       "FragmentType                                     y                          b   \n",
       "FragmentCharge                                   1                          1   \n",
       "FragmentSeriesNumber                             0                          1   \n",
       "FragmentLossType                               NaN                        NaN   \n",
       "Run                      Symb_Proteome_DIA_RAW_S05  Symb_Proteome_DIA_RAW_S05   \n",
       "\n",
       "                                              987   \n",
       "PrecursorMz                                738.038  \n",
       "ProductMz                                  1310.62  \n",
       "Annotation                                   y11^1  \n",
       "ProteinId                                   P13645  \n",
       "GeneName                                     KRT10  \n",
       "PeptideSequence                SKELTTEIDNNIEQISSYK  \n",
       "ModifiedPeptideSequence        SKELTTEIDNNIEQISSYK  \n",
       "PrecursorCharge                                  3  \n",
       "LibraryIntensity                           499.282  \n",
       "NormalizedRetentionTime                    83.1712  \n",
       "PrecursorIonMobility                           NaN  \n",
       "FragmentType                                     y  \n",
       "FragmentCharge                                   1  \n",
       "FragmentSeriesNumber                             1  \n",
       "FragmentLossType                               NaN  \n",
       "Run                      Symb_Proteome_DIA_RAW_S05  \n",
       "\n",
       "[16 rows x 30 columns]"
      ]
     },
     "execution_count": 156,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_one.sort_values([\"ProteinId\", \"Run\"]).tail(30).T"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Not using Fragpipe's labling for each replicate in each group -> One collective spectral library"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Run diann with the collective spectral library on all raw files."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [],
   "source": [
    "preSyntx = [diann, \"--lib\", \"/home/ekvall/Desktop/sample_A_and_S_one_lib/library.tsv\", \"--out\", \"/home/ekvall/Desktop/sample_A_and_S_one_lib/diann_out.tsv\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [],
   "source": [
    "def runAll(short_name_list, exp_name_list):\n",
    "    files = list()\n",
    "    pre_path = \"/sdd/proteomics/DATA/PXD014076/\"\n",
    "    for short_name, exp_name in zip(short_name_list, exp_name_list):\n",
    "        for r_n in range(1,6):\n",
    "            file = pre_path + \"{}/mzML/{}_Proteome_DIA_RAW_{}0{}.mzML\".format(exp_name, exp_name, short_name, r_n)\n",
    "            \n",
    "            files += [\"--f\", file]\n",
    "    return files\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [],
   "source": [
    "files = runAll([\"S\", \"A\"],[\"Symb\", \"Aposymb\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [],
   "source": [
    "syntax = \" \".join(preSyntx + files)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "DIA-NN 1.7.12 (Data Independent Acquisition by Neural Networks)\n",
      "Compiled on Feb 13 2021 17:30:41\n",
      "Current date and time: Mon Mar  1 18:03:28 2021\n",
      "Logical CPU cores: 8\n",
      "\n",
      "10 files will be processed\n",
      "[0:00] Loading spectral library /home/ekvall/Desktop/sample_A_and_S_one_lib/library.tsv\n",
      "[0:00] Finding proteotypic peptides (assuming that the list of UniProt ids provided for each peptide is complete)\n",
      "[0:00] Spectral library loaded: 222 protein isoforms, 222 protein groups and 253 precursors in 253 elution groups.\n",
      "[0:00] Initialising library\n",
      "[0:00] Saving the library to /home/ekvall/Desktop/sample_A_and_S_one_lib/library.tsv.speclib\n",
      "\n",
      "[0:00] File #1/10\n",
      "[0:00] Loading run /sdd/proteomics/DATA/PXD014076/Symb/mzML/Symb_Proteome_DIA_RAW_S01.mzML\n",
      "[0:26] 253 library precursors are potentially detectable\n",
      "[0:26] Processing...\n",
      "[0:30] RT window set to 2.36785\n",
      "[0:30] Peak width: 4.869\n",
      "[0:30] Scan window radius set to 10\n",
      "[0:34] Optimised mass accuracy: 16.4517 ppm\n",
      "[0:34] Removing interfering precursors\n",
      "[0:35] Training the neural network: 247 targets, 223 decoys\n",
      "[0:35] Number of IDs at 0.01 FDR: 237\n",
      "[0:35] Too low number of IDs with NNs: reverting to the linear classifier\n",
      "[0:35] Number of IDs at 0.01 FDR: 207\n",
      "[0:35] Calculating protein q-values\n",
      "[0:35] Number of protein isoforms identified at 1% FDR: 183 (precursor-level), 179 (protein-level) (inference performed using proteotypic peptides only)\n",
      "[0:35] Quantification\n",
      "[0:35] Quantification information saved to /sdd/proteomics/DATA/PXD014076/Symb/mzML/Symb_Proteome_DIA_RAW_S01.mzML.quant.\n",
      "\n",
      "[0:35] File #2/10\n",
      "[0:35] Loading run /sdd/proteomics/DATA/PXD014076/Symb/mzML/Symb_Proteome_DIA_RAW_S02.mzML\n",
      "[1:01] 253 library precursors are potentially detectable\n",
      "[1:01] Processing...\n",
      "[1:04] RT window set to 2.37119\n",
      "[1:05] Removing interfering precursors\n",
      "[1:05] Training the neural network: 250 targets, 221 decoys\n",
      "[1:05] Number of IDs at 0.01 FDR: 240\n",
      "[1:05] Calculating protein q-values\n",
      "[1:05] Number of protein isoforms identified at 1% FDR: 212 (precursor-level), 211 (protein-level) (inference performed using proteotypic peptides only)\n",
      "[1:05] Quantification\n",
      "[1:05] Quantification information saved to /sdd/proteomics/DATA/PXD014076/Symb/mzML/Symb_Proteome_DIA_RAW_S02.mzML.quant.\n",
      "\n",
      "[1:05] File #3/10\n",
      "[1:05] Loading run /sdd/proteomics/DATA/PXD014076/Symb/mzML/Symb_Proteome_DIA_RAW_S03.mzML\n",
      "[1:31] 253 library precursors are potentially detectable\n",
      "[1:31] Processing...\n",
      "[1:35] RT window set to 2.38191\n",
      "[1:35] Removing interfering precursors\n",
      "[1:35] Training the neural network: 250 targets, 222 decoys\n",
      "[1:35] Number of IDs at 0.01 FDR: 241\n",
      "[1:35] Calculating protein q-values\n",
      "[1:35] Number of protein isoforms identified at 1% FDR: 212 (precursor-level), 158 (protein-level) (inference performed using proteotypic peptides only)\n",
      "[1:35] Quantification\n",
      "[1:35] Quantification information saved to /sdd/proteomics/DATA/PXD014076/Symb/mzML/Symb_Proteome_DIA_RAW_S03.mzML.quant.\n",
      "\n",
      "[1:35] File #4/10\n",
      "[1:35] Loading run /sdd/proteomics/DATA/PXD014076/Symb/mzML/Symb_Proteome_DIA_RAW_S04.mzML\n",
      "[2:00] 253 library precursors are potentially detectable\n",
      "[2:00] Processing...\n",
      "[2:03] RT window set to 2.40764\n",
      "[2:03] Removing interfering precursors\n",
      "[2:03] Training the neural network: 225 targets, 199 decoys\n",
      "[2:03] Number of IDs at 0.01 FDR: 197\n",
      "[2:03] Calculating protein q-values\n",
      "[2:03] Number of protein isoforms identified at 1% FDR: 179 (precursor-level), 16 (protein-level) (inference performed using proteotypic peptides only)\n",
      "[2:03] Quantification\n",
      "[2:03] Quantification information saved to /sdd/proteomics/DATA/PXD014076/Symb/mzML/Symb_Proteome_DIA_RAW_S04.mzML.quant.\n",
      "\n",
      "[2:03] File #5/10\n",
      "[2:03] Loading run /sdd/proteomics/DATA/PXD014076/Symb/mzML/Symb_Proteome_DIA_RAW_S05.mzML\n",
      "[2:28] 253 library precursors are potentially detectable\n",
      "[2:28] Processing...\n",
      "[2:32] RT window set to 2.38295\n",
      "[2:32] Removing interfering precursors\n",
      "[2:32] Training the neural network: 248 targets, 212 decoys\n",
      "[2:32] Number of IDs at 0.01 FDR: 224\n",
      "[2:32] Too low number of IDs with NNs: reverting to the linear classifier\n",
      "[2:32] Number of IDs at 0.01 FDR: 220\n",
      "[2:32] Calculating protein q-values\n",
      "[2:32] Number of protein isoforms identified at 1% FDR: 197 (precursor-level), 25 (protein-level) (inference performed using proteotypic peptides only)\n",
      "[2:32] Quantification\n",
      "[2:32] Quantification information saved to /sdd/proteomics/DATA/PXD014076/Symb/mzML/Symb_Proteome_DIA_RAW_S05.mzML.quant.\n",
      "\n",
      "[2:32] File #6/10\n",
      "[2:32] Loading run /sdd/proteomics/DATA/PXD014076/Aposymb/mzML/Aposymb_Proteome_DIA_RAW_A01.mzML\n",
      "[2:56] 253 library precursors are potentially detectable\n",
      "[2:56] Processing...\n",
      "[3:00] RT window set to 2.37209\n",
      "[3:00] Removing interfering precursors\n",
      "[3:00] Training the neural network: 248 targets, 225 decoys\n",
      "[3:00] Number of IDs at 0.01 FDR: 229\n",
      "[3:00] Too low number of IDs with NNs: reverting to the linear classifier\n",
      "[3:00] Number of IDs at 0.01 FDR: 221\n",
      "[3:00] Calculating protein q-values\n",
      "[3:00] Number of protein isoforms identified at 1% FDR: 197 (precursor-level), 182 (protein-level) (inference performed using proteotypic peptides only)\n",
      "[3:00] Quantification\n",
      "[3:00] Quantification information saved to /sdd/proteomics/DATA/PXD014076/Aposymb/mzML/Aposymb_Proteome_DIA_RAW_A01.mzML.quant.\n",
      "\n",
      "[3:00] File #7/10\n",
      "[3:00] Loading run /sdd/proteomics/DATA/PXD014076/Aposymb/mzML/Aposymb_Proteome_DIA_RAW_A02.mzML\n",
      "[3:25] 253 library precursors are potentially detectable\n",
      "[3:25] Processing...\n",
      "[3:28] RT window set to 2.3727\n",
      "[3:29] Removing interfering precursors\n",
      "[3:29] Training the neural network: 235 targets, 224 decoys\n",
      "[3:29] Number of IDs at 0.01 FDR: 222\n",
      "[3:29] Calculating protein q-values\n",
      "[3:29] Number of protein isoforms identified at 1% FDR: 201 (precursor-level), 201 (protein-level) (inference performed using proteotypic peptides only)\n",
      "[3:29] Quantification\n",
      "[3:29] Quantification information saved to /sdd/proteomics/DATA/PXD014076/Aposymb/mzML/Aposymb_Proteome_DIA_RAW_A02.mzML.quant.\n",
      "\n",
      "[3:29] File #8/10\n",
      "[3:29] Loading run /sdd/proteomics/DATA/PXD014076/Aposymb/mzML/Aposymb_Proteome_DIA_RAW_A03.mzML\n",
      "[3:56] 253 library precursors are potentially detectable\n",
      "[3:56] Processing...\n",
      "[3:59] RT window set to 2.36493\n",
      "[4:00] Removing interfering precursors\n",
      "[4:00] Training the neural network: 246 targets, 232 decoys\n",
      "[4:00] Number of IDs at 0.01 FDR: 238\n",
      "[4:00] Too low number of IDs with NNs: reverting to the linear classifier\n",
      "[4:00] Number of IDs at 0.01 FDR: 226\n",
      "[4:00] Calculating protein q-values\n",
      "[4:00] Number of protein isoforms identified at 1% FDR: 199 (precursor-level), 199 (protein-level) (inference performed using proteotypic peptides only)\n",
      "[4:00] Quantification\n",
      "[4:00] Quantification information saved to /sdd/proteomics/DATA/PXD014076/Aposymb/mzML/Aposymb_Proteome_DIA_RAW_A03.mzML.quant.\n",
      "\n",
      "[4:00] File #9/10\n",
      "[4:00] Loading run /sdd/proteomics/DATA/PXD014076/Aposymb/mzML/Aposymb_Proteome_DIA_RAW_A04.mzML\n",
      "[4:26] 253 library precursors are potentially detectable\n",
      "[4:26] Processing...\n",
      "[4:29] RT window set to 2.31614\n",
      "[4:29] Removing interfering precursors\n",
      "[4:29] Training the neural network: 244 targets, 227 decoys\n",
      "[4:30] Number of IDs at 0.01 FDR: 229\n",
      "[4:30] Calculating protein q-values\n",
      "[4:30] Number of protein isoforms identified at 1% FDR: 202 (precursor-level), 140 (protein-level) (inference performed using proteotypic peptides only)\n",
      "[4:30] Quantification\n",
      "[4:30] Quantification information saved to /sdd/proteomics/DATA/PXD014076/Aposymb/mzML/Aposymb_Proteome_DIA_RAW_A04.mzML.quant.\n",
      "\n",
      "[4:30] File #10/10\n",
      "[4:30] Loading run /sdd/proteomics/DATA/PXD014076/Aposymb/mzML/Aposymb_Proteome_DIA_RAW_A05.mzML\n",
      "[4:55] 253 library precursors are potentially detectable\n",
      "[4:55] Processing...\n",
      "[4:59] RT window set to 2.36709\n",
      "[4:59] Removing interfering precursors\n",
      "[4:59] Training the neural network: 242 targets, 219 decoys\n",
      "[4:59] Number of IDs at 0.01 FDR: 222\n",
      "[4:59] Too low number of IDs with NNs: reverting to the linear classifier\n",
      "[4:59] Number of IDs at 0.01 FDR: 214\n",
      "[4:59] Calculating protein q-values\n",
      "[4:59] Number of protein isoforms identified at 1% FDR: 189 (precursor-level), 22 (protein-level) (inference performed using proteotypic peptides only)\n",
      "[4:59] Quantification\n",
      "[5:00] Quantification information saved to /sdd/proteomics/DATA/PXD014076/Aposymb/mzML/Aposymb_Proteome_DIA_RAW_A05.mzML.quant.\n",
      "\n",
      "[5:00] Cross-run analysis\n",
      "[5:00] Reading quantification information: 10 files\n",
      "[5:00] Quantifying peptides\n",
      "[5:00] Assembling protein groups\n",
      "[5:00] Quantifying proteins\n",
      "[5:00] Calculating q-values for protein and gene groups\n",
      "[5:00] Writing report\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[5:00] Report saved to /home/ekvall/Desktop/sample_A_and_S_one_lib/diann_out.tsv.\r\n",
      "[5:00] Stats report saved to /home/ekvall/Desktop/sample_A_and_S_one_lib/diann_out.stats.tsv\r\n",
      "[5:00] Writing gene report\r\n",
      "[5:00] Gene report saved to report.genes.tsv.\r\n",
      "[5:00] Log saved to /home/ekvall/Desktop/sample_A_and_S_one_lib/diann_out.log.txt\r\n",
      "Finished\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "!{syntax}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Load diann output for the collective spectral library with all the raw files."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {},
   "outputs": [],
   "source": [
    "diann_all = pd.read_csv(\"/home/ekvall/Desktop/sample_A_and_S_one_lib/diann_out.tsv\", sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(2217, 42)"
      ]
     },
     "execution_count": 144,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "diann_all.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['File.Name', 'Run', 'Protein.Group', 'Protein.Ids', 'Protein.Names',\n",
       "       'Genes', 'PG.Quantity', 'PG.Normalised', 'Genes.Quantity',\n",
       "       'Genes.Normalised', 'Genes.MaxLFQ', 'Genes.MaxLFQ.Unique',\n",
       "       'Modified.Sequence', 'Stripped.Sequence', 'Precursor.Id',\n",
       "       'Precursor.Charge', 'Q.Value', 'Protein.Q.Value', 'PG.Q.Value',\n",
       "       'GG.Q.Value', 'Proteotypic', 'Precursor.Quantity',\n",
       "       'Precursor.Normalised', 'Label.Ratio', 'Quantity.Quality', 'RT',\n",
       "       'RT.Start', 'RT.Stop', 'iRT', 'Predicted.RT', 'Predicted.iRT',\n",
       "       'Lib.Q.Value', 'Ms1.Profile.Corr', 'Ms1.Area', 'Evidence', 'CScore',\n",
       "       'Decoy.Evidence', 'Decoy.CScore', 'Fragment.Quant.Raw',\n",
       "       'Fragment.Quant.Corrected', 'Fragment.Correlations', 'MS2.Scan'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 145,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "diann_all.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>1659</th>\n",
       "      <th>1651</th>\n",
       "      <th>1652</th>\n",
       "      <th>1653</th>\n",
       "      <th>1654</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>File.Name</th>\n",
       "      <td>/sdd/proteomics/DATA/PXD014076/Aposymb/mzML/Ap...</td>\n",
       "      <td>/sdd/proteomics/DATA/PXD014076/Symb/mzML/Symb_...</td>\n",
       "      <td>/sdd/proteomics/DATA/PXD014076/Symb/mzML/Symb_...</td>\n",
       "      <td>/sdd/proteomics/DATA/PXD014076/Symb/mzML/Symb_...</td>\n",
       "      <td>/sdd/proteomics/DATA/PXD014076/Symb/mzML/Symb_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Run</th>\n",
       "      <td>Aposymb_Proteome_DIA_RAW_A05</td>\n",
       "      <td>Symb_Proteome_DIA_RAW_S01</td>\n",
       "      <td>Symb_Proteome_DIA_RAW_S02</td>\n",
       "      <td>Symb_Proteome_DIA_RAW_S03</td>\n",
       "      <td>Symb_Proteome_DIA_RAW_S05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Protein.Group</th>\n",
       "      <td>P13645</td>\n",
       "      <td>P13645</td>\n",
       "      <td>P13645</td>\n",
       "      <td>P13645</td>\n",
       "      <td>P13645</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Protein.Ids</th>\n",
       "      <td>P13645</td>\n",
       "      <td>P13645</td>\n",
       "      <td>P13645</td>\n",
       "      <td>P13645</td>\n",
       "      <td>P13645</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Protein.Names</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genes</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PG.Quantity</th>\n",
       "      <td>1.43206e+06</td>\n",
       "      <td>4.09343e+06</td>\n",
       "      <td>7.79829e+06</td>\n",
       "      <td>4.53727e+06</td>\n",
       "      <td>1.80206e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PG.Normalised</th>\n",
       "      <td>1.49345e+06</td>\n",
       "      <td>3.31164e+06</td>\n",
       "      <td>7.34234e+06</td>\n",
       "      <td>3.7743e+06</td>\n",
       "      <td>2.87377e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genes.Quantity</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genes.Normalised</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genes.MaxLFQ</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genes.MaxLFQ.Unique</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Modified.Sequence</th>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Stripped.Sequence</th>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Precursor.Id</th>\n",
       "      <td>SKELTTEIDNNIEQISSYK3</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK3</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK3</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK3</td>\n",
       "      <td>SKELTTEIDNNIEQISSYK3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Precursor.Charge</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Q.Value</th>\n",
       "      <td>0.00480769</td>\n",
       "      <td>0.00487805</td>\n",
       "      <td>0.00833333</td>\n",
       "      <td>0.00568182</td>\n",
       "      <td>0.00454545</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Protein.Q.Value</th>\n",
       "      <td>0.010582</td>\n",
       "      <td>0.00546448</td>\n",
       "      <td>0.00473934</td>\n",
       "      <td>0.00943396</td>\n",
       "      <td>0.0149254</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PG.Q.Value</th>\n",
       "      <td>0.010582</td>\n",
       "      <td>0.00546448</td>\n",
       "      <td>0.00473934</td>\n",
       "      <td>0.00943396</td>\n",
       "      <td>0.0149254</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GG.Q.Value</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Proteotypic</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Precursor.Quantity</th>\n",
       "      <td>1.43206e+06</td>\n",
       "      <td>4.09343e+06</td>\n",
       "      <td>7.79829e+06</td>\n",
       "      <td>4.53727e+06</td>\n",
       "      <td>1.80206e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Precursor.Normalised</th>\n",
       "      <td>1.49345e+06</td>\n",
       "      <td>3.31164e+06</td>\n",
       "      <td>7.34234e+06</td>\n",
       "      <td>3.7743e+06</td>\n",
       "      <td>2.87377e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Label.Ratio</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Quantity.Quality</th>\n",
       "      <td>0.958421</td>\n",
       "      <td>0.98528</td>\n",
       "      <td>0.995519</td>\n",
       "      <td>0.987806</td>\n",
       "      <td>0.987005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RT</th>\n",
       "      <td>101.103</td>\n",
       "      <td>100.554</td>\n",
       "      <td>100.676</td>\n",
       "      <td>100.413</td>\n",
       "      <td>100.525</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RT.Start</th>\n",
       "      <td>100.896</td>\n",
       "      <td>100.231</td>\n",
       "      <td>100.41</td>\n",
       "      <td>100.147</td>\n",
       "      <td>100.298</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RT.Stop</th>\n",
       "      <td>101.309</td>\n",
       "      <td>100.765</td>\n",
       "      <td>100.885</td>\n",
       "      <td>100.677</td>\n",
       "      <td>100.749</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>iRT</th>\n",
       "      <td>84.9467</td>\n",
       "      <td>84.9467</td>\n",
       "      <td>84.9467</td>\n",
       "      <td>84.9467</td>\n",
       "      <td>84.9467</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Predicted.RT</th>\n",
       "      <td>101.118</td>\n",
       "      <td>100.411</td>\n",
       "      <td>100.545</td>\n",
       "      <td>100.421</td>\n",
       "      <td>100.462</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Predicted.iRT</th>\n",
       "      <td>84.9422</td>\n",
       "      <td>85.0837</td>\n",
       "      <td>85.0729</td>\n",
       "      <td>84.9409</td>\n",
       "      <td>85.0153</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Lib.Q.Value</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Ms1.Profile.Corr</th>\n",
       "      <td>-0.192866</td>\n",
       "      <td>0.0365489</td>\n",
       "      <td>0.426074</td>\n",
       "      <td>0.124971</td>\n",
       "      <td>-0.0590517</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Ms1.Area</th>\n",
       "      <td>4.94596e+06</td>\n",
       "      <td>1.35927e+07</td>\n",
       "      <td>2.49769e+07</td>\n",
       "      <td>1.48613e+07</td>\n",
       "      <td>5.92688e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Evidence</th>\n",
       "      <td>4.44363</td>\n",
       "      <td>5.39201</td>\n",
       "      <td>5.8035</td>\n",
       "      <td>5.5708</td>\n",
       "      <td>5.27646</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CScore</th>\n",
       "      <td>3.10776</td>\n",
       "      <td>6.10031</td>\n",
       "      <td>0.969757</td>\n",
       "      <td>0.966411</td>\n",
       "      <td>9.64089</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Decoy.Evidence</th>\n",
       "      <td>1.49355</td>\n",
       "      <td>1.47881</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Decoy.CScore</th>\n",
       "      <td>-5.04348</td>\n",
       "      <td>-8.07795</td>\n",
       "      <td>-1e+07</td>\n",
       "      <td>-1e+07</td>\n",
       "      <td>-1e+07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Fragment.Quant.Raw</th>\n",
       "      <td>781489;445996;843890;249431;82302.1;244547;</td>\n",
       "      <td>2.19266e+06;1.19193e+06;1.10931e+06;708836;868...</td>\n",
       "      <td>4.16989e+06;2.38503e+06;1.36998e+06;1.31577e+0...</td>\n",
       "      <td>2.49903e+06;1.39561e+06;1.10975e+06;649869;716...</td>\n",
       "      <td>998131;571300;577231;269157;235569;388746;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Fragment.Quant.Corrected</th>\n",
       "      <td>758570;424055;461100;249431;0;215819;</td>\n",
       "      <td>2.19266e+06;1.19193e+06;790575;708836;738882;8...</td>\n",
       "      <td>4.0975e+06;2.38503e+06;1.18888e+06;1.31577e+06...</td>\n",
       "      <td>2.49903e+06;1.39561e+06;800419;642634;675106;8...</td>\n",
       "      <td>961604;571300;420320;269157;110775;364326;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Fragment.Correlations</th>\n",
       "      <td>0.960963;0.938947;0.433654;0.983797;-0.530827;...</td>\n",
       "      <td>0.985602;0.9772;0.505779;0.997869;0.821892;0.9...</td>\n",
       "      <td>0.994368;0.996231;0.932025;0.997814;0.95312;0....</td>\n",
       "      <td>0.996585;0.982349;0.596744;0.965519;0.812165;0...</td>\n",
       "      <td>0.984828;0.997289;0.582973;0.972953;0.623506;0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MS2.Scan</th>\n",
       "      <td>57389</td>\n",
       "      <td>57901</td>\n",
       "      <td>57581</td>\n",
       "      <td>57901</td>\n",
       "      <td>56269</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                       1659  \\\n",
       "File.Name                 /sdd/proteomics/DATA/PXD014076/Aposymb/mzML/Ap...   \n",
       "Run                                            Aposymb_Proteome_DIA_RAW_A05   \n",
       "Protein.Group                                                        P13645   \n",
       "Protein.Ids                                                          P13645   \n",
       "Protein.Names                                                           NaN   \n",
       "Genes                                                                   NaN   \n",
       "PG.Quantity                                                     1.43206e+06   \n",
       "PG.Normalised                                                   1.49345e+06   \n",
       "Genes.Quantity                                                          NaN   \n",
       "Genes.Normalised                                                        NaN   \n",
       "Genes.MaxLFQ                                                            NaN   \n",
       "Genes.MaxLFQ.Unique                                                     NaN   \n",
       "Modified.Sequence                                       SKELTTEIDNNIEQISSYK   \n",
       "Stripped.Sequence                                       SKELTTEIDNNIEQISSYK   \n",
       "Precursor.Id                                           SKELTTEIDNNIEQISSYK3   \n",
       "Precursor.Charge                                                          3   \n",
       "Q.Value                                                          0.00480769   \n",
       "Protein.Q.Value                                                    0.010582   \n",
       "PG.Q.Value                                                         0.010582   \n",
       "GG.Q.Value                                                                1   \n",
       "Proteotypic                                                               1   \n",
       "Precursor.Quantity                                              1.43206e+06   \n",
       "Precursor.Normalised                                            1.49345e+06   \n",
       "Label.Ratio                                                               0   \n",
       "Quantity.Quality                                                   0.958421   \n",
       "RT                                                                  101.103   \n",
       "RT.Start                                                            100.896   \n",
       "RT.Stop                                                             101.309   \n",
       "iRT                                                                 84.9467   \n",
       "Predicted.RT                                                        101.118   \n",
       "Predicted.iRT                                                       84.9422   \n",
       "Lib.Q.Value                                                               0   \n",
       "Ms1.Profile.Corr                                                  -0.192866   \n",
       "Ms1.Area                                                        4.94596e+06   \n",
       "Evidence                                                            4.44363   \n",
       "CScore                                                              3.10776   \n",
       "Decoy.Evidence                                                      1.49355   \n",
       "Decoy.CScore                                                       -5.04348   \n",
       "Fragment.Quant.Raw              781489;445996;843890;249431;82302.1;244547;   \n",
       "Fragment.Quant.Corrected              758570;424055;461100;249431;0;215819;   \n",
       "Fragment.Correlations     0.960963;0.938947;0.433654;0.983797;-0.530827;...   \n",
       "MS2.Scan                                                              57389   \n",
       "\n",
       "                                                                       1651  \\\n",
       "File.Name                 /sdd/proteomics/DATA/PXD014076/Symb/mzML/Symb_...   \n",
       "Run                                               Symb_Proteome_DIA_RAW_S01   \n",
       "Protein.Group                                                        P13645   \n",
       "Protein.Ids                                                          P13645   \n",
       "Protein.Names                                                           NaN   \n",
       "Genes                                                                   NaN   \n",
       "PG.Quantity                                                     4.09343e+06   \n",
       "PG.Normalised                                                   3.31164e+06   \n",
       "Genes.Quantity                                                          NaN   \n",
       "Genes.Normalised                                                        NaN   \n",
       "Genes.MaxLFQ                                                            NaN   \n",
       "Genes.MaxLFQ.Unique                                                     NaN   \n",
       "Modified.Sequence                                       SKELTTEIDNNIEQISSYK   \n",
       "Stripped.Sequence                                       SKELTTEIDNNIEQISSYK   \n",
       "Precursor.Id                                           SKELTTEIDNNIEQISSYK3   \n",
       "Precursor.Charge                                                          3   \n",
       "Q.Value                                                          0.00487805   \n",
       "Protein.Q.Value                                                  0.00546448   \n",
       "PG.Q.Value                                                       0.00546448   \n",
       "GG.Q.Value                                                                1   \n",
       "Proteotypic                                                               1   \n",
       "Precursor.Quantity                                              4.09343e+06   \n",
       "Precursor.Normalised                                            3.31164e+06   \n",
       "Label.Ratio                                                               0   \n",
       "Quantity.Quality                                                    0.98528   \n",
       "RT                                                                  100.554   \n",
       "RT.Start                                                            100.231   \n",
       "RT.Stop                                                             100.765   \n",
       "iRT                                                                 84.9467   \n",
       "Predicted.RT                                                        100.411   \n",
       "Predicted.iRT                                                       85.0837   \n",
       "Lib.Q.Value                                                               0   \n",
       "Ms1.Profile.Corr                                                  0.0365489   \n",
       "Ms1.Area                                                        1.35927e+07   \n",
       "Evidence                                                            5.39201   \n",
       "CScore                                                              6.10031   \n",
       "Decoy.Evidence                                                      1.47881   \n",
       "Decoy.CScore                                                       -8.07795   \n",
       "Fragment.Quant.Raw        2.19266e+06;1.19193e+06;1.10931e+06;708836;868...   \n",
       "Fragment.Quant.Corrected  2.19266e+06;1.19193e+06;790575;708836;738882;8...   \n",
       "Fragment.Correlations     0.985602;0.9772;0.505779;0.997869;0.821892;0.9...   \n",
       "MS2.Scan                                                              57901   \n",
       "\n",
       "                                                                       1652  \\\n",
       "File.Name                 /sdd/proteomics/DATA/PXD014076/Symb/mzML/Symb_...   \n",
       "Run                                               Symb_Proteome_DIA_RAW_S02   \n",
       "Protein.Group                                                        P13645   \n",
       "Protein.Ids                                                          P13645   \n",
       "Protein.Names                                                           NaN   \n",
       "Genes                                                                   NaN   \n",
       "PG.Quantity                                                     7.79829e+06   \n",
       "PG.Normalised                                                   7.34234e+06   \n",
       "Genes.Quantity                                                          NaN   \n",
       "Genes.Normalised                                                        NaN   \n",
       "Genes.MaxLFQ                                                            NaN   \n",
       "Genes.MaxLFQ.Unique                                                     NaN   \n",
       "Modified.Sequence                                       SKELTTEIDNNIEQISSYK   \n",
       "Stripped.Sequence                                       SKELTTEIDNNIEQISSYK   \n",
       "Precursor.Id                                           SKELTTEIDNNIEQISSYK3   \n",
       "Precursor.Charge                                                          3   \n",
       "Q.Value                                                          0.00833333   \n",
       "Protein.Q.Value                                                  0.00473934   \n",
       "PG.Q.Value                                                       0.00473934   \n",
       "GG.Q.Value                                                                1   \n",
       "Proteotypic                                                               1   \n",
       "Precursor.Quantity                                              7.79829e+06   \n",
       "Precursor.Normalised                                            7.34234e+06   \n",
       "Label.Ratio                                                               0   \n",
       "Quantity.Quality                                                   0.995519   \n",
       "RT                                                                  100.676   \n",
       "RT.Start                                                             100.41   \n",
       "RT.Stop                                                             100.885   \n",
       "iRT                                                                 84.9467   \n",
       "Predicted.RT                                                        100.545   \n",
       "Predicted.iRT                                                       85.0729   \n",
       "Lib.Q.Value                                                               0   \n",
       "Ms1.Profile.Corr                                                   0.426074   \n",
       "Ms1.Area                                                        2.49769e+07   \n",
       "Evidence                                                             5.8035   \n",
       "CScore                                                             0.969757   \n",
       "Decoy.Evidence                                                            0   \n",
       "Decoy.CScore                                                         -1e+07   \n",
       "Fragment.Quant.Raw        4.16989e+06;2.38503e+06;1.36998e+06;1.31577e+0...   \n",
       "Fragment.Quant.Corrected  4.0975e+06;2.38503e+06;1.18888e+06;1.31577e+06...   \n",
       "Fragment.Correlations     0.994368;0.996231;0.932025;0.997814;0.95312;0....   \n",
       "MS2.Scan                                                              57581   \n",
       "\n",
       "                                                                       1653  \\\n",
       "File.Name                 /sdd/proteomics/DATA/PXD014076/Symb/mzML/Symb_...   \n",
       "Run                                               Symb_Proteome_DIA_RAW_S03   \n",
       "Protein.Group                                                        P13645   \n",
       "Protein.Ids                                                          P13645   \n",
       "Protein.Names                                                           NaN   \n",
       "Genes                                                                   NaN   \n",
       "PG.Quantity                                                     4.53727e+06   \n",
       "PG.Normalised                                                    3.7743e+06   \n",
       "Genes.Quantity                                                          NaN   \n",
       "Genes.Normalised                                                        NaN   \n",
       "Genes.MaxLFQ                                                            NaN   \n",
       "Genes.MaxLFQ.Unique                                                     NaN   \n",
       "Modified.Sequence                                       SKELTTEIDNNIEQISSYK   \n",
       "Stripped.Sequence                                       SKELTTEIDNNIEQISSYK   \n",
       "Precursor.Id                                           SKELTTEIDNNIEQISSYK3   \n",
       "Precursor.Charge                                                          3   \n",
       "Q.Value                                                          0.00568182   \n",
       "Protein.Q.Value                                                  0.00943396   \n",
       "PG.Q.Value                                                       0.00943396   \n",
       "GG.Q.Value                                                                1   \n",
       "Proteotypic                                                               1   \n",
       "Precursor.Quantity                                              4.53727e+06   \n",
       "Precursor.Normalised                                             3.7743e+06   \n",
       "Label.Ratio                                                               0   \n",
       "Quantity.Quality                                                   0.987806   \n",
       "RT                                                                  100.413   \n",
       "RT.Start                                                            100.147   \n",
       "RT.Stop                                                             100.677   \n",
       "iRT                                                                 84.9467   \n",
       "Predicted.RT                                                        100.421   \n",
       "Predicted.iRT                                                       84.9409   \n",
       "Lib.Q.Value                                                               0   \n",
       "Ms1.Profile.Corr                                                   0.124971   \n",
       "Ms1.Area                                                        1.48613e+07   \n",
       "Evidence                                                             5.5708   \n",
       "CScore                                                             0.966411   \n",
       "Decoy.Evidence                                                            0   \n",
       "Decoy.CScore                                                         -1e+07   \n",
       "Fragment.Quant.Raw        2.49903e+06;1.39561e+06;1.10975e+06;649869;716...   \n",
       "Fragment.Quant.Corrected  2.49903e+06;1.39561e+06;800419;642634;675106;8...   \n",
       "Fragment.Correlations     0.996585;0.982349;0.596744;0.965519;0.812165;0...   \n",
       "MS2.Scan                                                              57901   \n",
       "\n",
       "                                                                       1654  \n",
       "File.Name                 /sdd/proteomics/DATA/PXD014076/Symb/mzML/Symb_...  \n",
       "Run                                               Symb_Proteome_DIA_RAW_S05  \n",
       "Protein.Group                                                        P13645  \n",
       "Protein.Ids                                                          P13645  \n",
       "Protein.Names                                                           NaN  \n",
       "Genes                                                                   NaN  \n",
       "PG.Quantity                                                     1.80206e+06  \n",
       "PG.Normalised                                                   2.87377e+06  \n",
       "Genes.Quantity                                                          NaN  \n",
       "Genes.Normalised                                                        NaN  \n",
       "Genes.MaxLFQ                                                            NaN  \n",
       "Genes.MaxLFQ.Unique                                                     NaN  \n",
       "Modified.Sequence                                       SKELTTEIDNNIEQISSYK  \n",
       "Stripped.Sequence                                       SKELTTEIDNNIEQISSYK  \n",
       "Precursor.Id                                           SKELTTEIDNNIEQISSYK3  \n",
       "Precursor.Charge                                                          3  \n",
       "Q.Value                                                          0.00454545  \n",
       "Protein.Q.Value                                                   0.0149254  \n",
       "PG.Q.Value                                                        0.0149254  \n",
       "GG.Q.Value                                                                1  \n",
       "Proteotypic                                                               1  \n",
       "Precursor.Quantity                                              1.80206e+06  \n",
       "Precursor.Normalised                                            2.87377e+06  \n",
       "Label.Ratio                                                               0  \n",
       "Quantity.Quality                                                   0.987005  \n",
       "RT                                                                  100.525  \n",
       "RT.Start                                                            100.298  \n",
       "RT.Stop                                                             100.749  \n",
       "iRT                                                                 84.9467  \n",
       "Predicted.RT                                                        100.462  \n",
       "Predicted.iRT                                                       85.0153  \n",
       "Lib.Q.Value                                                               0  \n",
       "Ms1.Profile.Corr                                                 -0.0590517  \n",
       "Ms1.Area                                                        5.92688e+06  \n",
       "Evidence                                                            5.27646  \n",
       "CScore                                                              9.64089  \n",
       "Decoy.Evidence                                                            0  \n",
       "Decoy.CScore                                                         -1e+07  \n",
       "Fragment.Quant.Raw               998131;571300;577231;269157;235569;388746;  \n",
       "Fragment.Quant.Corrected         961604;571300;420320;269157;110775;364326;  \n",
       "Fragment.Correlations     0.984828;0.997289;0.582973;0.972953;0.623506;0...  \n",
       "MS2.Scan                                                              56269  "
      ]
     },
     "execution_count": 150,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "diann_all.sort_values([\"Protein.Ids\", \"Run\"]).tail(5).T"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python (terran3)",
   "language": "python",
   "name": "terran3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
